### DISCUSSION TITLE
Problemas test e2e & CI & microservicios

### DISCUSSION THREAD

**Discussion by gs-Manuel** [Arquisoft/faq#63] [Q&A]
Estamos añadiendo tests e2e a nuestra aplicación con las herramientas que ya vienen configuradas en el proyecto de prueba (*jest-puppeteer, start-server-and-test, javascript, monogdb*). Nuestro problema reside en que empleamos mariadb para cada microservicio y por lo que he estado buscando, no he encontrado ningún paquete que ejecute una BBDD en memoria, por lo que en CI no podría funcionar ya que no puede acceder al contenedor docker con la base de datos  mariadb.

Al ejecutar `npm run build` y `npm run test:e2e` e intentar acceder a los microservicios solo permite acceder a aquellos que no tienen base de datos como webapp, pero los que si la tienen están "caídos" como auth_service.

¿Alguna solución?

Gracias, un saludo.

**Comment by pglez82** [comment]
¿Y como habéis hecho para los tests unitarios? En los unitarios a mi se me ocurre mockear el acceso a la base de datos, con algo de este estilo: https://stackoverflow.com/a/10124424/2828454 (nunca lo he usado pero creo que podría servir).

Para los e2e deberíais de usar una base de datos real. Podríais lanzar mariadb a la vez que levantáis todos los servicios para testear y tener una bbdd para pruebas (diferente a la de "producción"), que antes de cada test, se limpiaría (o se inicializaría con algo en las tablas que pudiese necesitar el test en concreto).
